### PR TITLE
DAOS-7526 test: Adding io & system admin test

### PR DESCRIPTION
**Pull request by saurabhtandan**
Adding io & basic system admin test to the deployment
directory to check happy path io tests and system admin
activities.

Test-tag: pr iosysadmin

Signed-off-by: Saurabh Tandan <saurabh.tandan@intel.com>

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/deployment/io_sys_admin.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        for idx in range(1, 4):
+            self.add_pool_qty(1, namespace="/run/pool_{}/".format(idx), create=False)
+            PoolTestBase.check_pool_creation(self, 60)
+            self.pool[-1].connect()
+            self.add_container_qty(1, self.pool[-1],
+                                   namespace="/run/container_{}/".format(idx), create=False)
+            for _, container in enumerate(self.container):
+                container.create()
+                daos.container_set_owner(self.pool[-1].uuid, self.container[-1].uuid,
+                                         new_test_user, new_test_group)
```
Comment (phender) [inline src/tests/ftest/deployment/io_sys_admin.py:65]: For 3 containers per pool:
```suggestion
            for cont_idx in range(1, 4):
                self.add_container_qty(1, self.pool[-1], namespace="/run/container_{}/".format(cont_idx))
                daos.container_set_owner(self.pool[-1].uuid, self.container[-1].uuid,
                                         new_test_user, new_test_group)
```

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/deployment/io_sys_admin.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    def get_free_space(self):
+        """Display pool free space."""
+        free_space = []
+        scm_free_space = self.pool.get_pool_free_space("scm")
+        free_space.append(scm_free_space)
+        nvme_free_space = self.pool.get_pool_free_space("nvme")
+        free_space.append(nvme_free_space)
+        self.log.info("Free space [SCM, Nvme]: [%s, %s]", scm_free_space, nvme_free_space)
+
+        return free_space
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/deployment/io_sys_admin.py]: Done

File: src/tests/ftest/deployment/io_sys_admin.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        """
+        Test Description: Performs tests to generate large data sets over
+                          various middleware, perform various system admin
+                          operations, datamover operations.
+        :avocado: tags=hw,large
+        :avocado: tags=deployment,iosysadmin
+        """
+        # local param
+#        ec_ior_flags = self.params.get("ec_ior_flags", "/run/ior/*")
+#        hdf5_plugin_path = self.params.get("plugin_path", '/run/hdf5_vol/*')
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/deployment/io_sys_admin.py]: Done

File: src/tests/ftest/deployment/io_sys_admin.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+            self.container = None
+            self.destroy_pools(self.pool)
+            self.pool = None
+
+        # dmg storage scan
+        dmg.storage_scan()
+        dmg.system_query()
+        dmg.system_leader_query()
+
+        # wirte large data sets
```
[the remark this answers is not in the document]
Comment (saurabhtandan) [reply src/tests/ftest/deployment/io_sys_admin.py]: Done

File: src/tests/ftest/deployment/io_sys_admin.py (hunk 1 above)
Comment (saurabhtandan) [reply src/tests/ftest/deployment/io_sys_admin.py]: Done

**Comment by saurabhtandan** [comment]
The PR was run with 'pr' tag earlier
And the last run was run for just the 'iosysadmin' tag which is expected to fail with known issue of DAOS-9218

File: src/tests/ftest/util/file_count_test_base.py [hunk 5]
```diff
@@ -33,13 +33,14 @@ def get_rf(oclass):
...
 # pylint: disable=attribute-defined-outside-init
-class FileCountTestBase(MdtestBase, IorTestBase):
+class FileCountTestBase(IorTestBase, MdtestBase):
     # pylint: disable=too-many-ancestors
     """Test class Description: Runs IOR and MDTEST to create specified number of files.
 
     :avocado: recursive
     """
 
+
```
Comment (phender) [inline src/tests/ftest/util/file_count_test_base.py:43]: The pydocstyle recommendation is single-spacing between methods inside a class and double-spacing between classes or standalone methods in a file.
```suggestion
```

**Review by ravalsam — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Pull request merged**